frontend/src/metabase-lib/Question.ts

```diff
@@ -1327,216 +1323,31 @@ class QuestionInner {
-            ];
-          case "sum":
-            return [
-              t`Sum of `,
-              this._getFieldName(tableMetadata, aggregation[1], options),
-            ];
-          case "cum-sum":
-            return [
-              t`Cumulative sum of `,
-              this._getFieldName(tableMetadata, aggregation[1], options),
-            ];
-          case "max":
-            return [
-              t`Maximum of `,
-              this._getFieldName(tableMetadata, aggregation[1], options),
-            ];
-          case "min":
-            return [
-              t`Minimum of `,
-              this._getFieldName(tableMetadata, aggregation[1], options),
-            ];
-          default:
-            return [formatExpression(aggregation, { tableMetadata })];
-        }
-      }),
-      // TODO: This is untranslated. See if there's an i18n-friendly way to do a comma-separated list.
-      "and",
-    );
-  }
-
-  private _getBreakoutDescription(tableMetadata, { breakout }, options) {
-    if (breakout && breakout.length > 0) {
-      return [
-        t`Grouped by `,
-        DESCRIPTION.joinList(
-          breakout.map(b => this._getFieldName(tableMetadata, b, options)),
-          // TODO: This is untranslated. See if there's an i18n-friendly way to do a comma-separated list.
-          " and ",
-        ),
-      ];
-    }
-  }
-
-  private _getFilterDescription(tableMetadata, query, options) {
-    // getFilters returns list of filters without the implied "and"
-    // TODO: This is untranslated. See if there's an i18n-friendly way to do a comma-separated list.
-    const filters = ["and"].concat(QUERY.getFilters(query));
-    if (filters && filters.length > 1) {
-      return [
-        t`Filtered by `,
-        this._getFilterClauseDescription(tableMetadata, filters, options),
-      ];
-    }
+  _getMLv2Query(metadata = this._metadata) {
+    return MLv2.query(this.databaseId(), metadata, this.datasetQuery());
   }
 
-  private _getFilterClauseDescription(tableMetadata, filter, options) {
-    if (filter[0] === "and" || filter[0] === "or") {
-      const clauses = filter
-        .slice(1)
-        .map(f => this._getFilterClauseDescription(tableMetadata, f, options));
-      return DESCRIPTION.conjunctList(clauses, filter[0].toLowerCase());
-    } else if (filter[0] === "segment") {
-      const segment = _.findWhere(tableMetadata.segments, { id: filter[1] });
-      return segment ? segment.name : "[Unknown Segment]";
-    } else if (filter[0] === "between" && filter[1][0] === "+") {
-      return this._getFieldName(tableMetadata, filter[1][1], options);
-    } else {
-      return this._getFieldName(tableMetadata, filter[1], options);
-    }
-  }
+  generateQueryDescription(tableMetadata) {
+    let metadata =
+      tableMetadata?.id != null
+        ? assocIn(
+            this._metadata,
+            ["tables", String(tableMetadata.id)],
+            tableMetadata,
+          )
+        : this._metadata;
```

Comment:
I'm not sure why we need to modify the metadata here and below now?

Reply:
MLv2 currently wants queries to be initialized with complete metadata like `this._metadata`, not just metadata for a single table. So I'm just merging the metadata for the individual table into `this._metadata`. 

I suppose we could rework things so we can initialize a query with just the metadata for a single table but it seems like query descriptions would break if you happened to be filtering on a field from a different table, so it didn't make a ton of sense to relax that rule honestly

Reply:
@camsaul I see, thank you. What if we rework the caller so it actually provides the full `Metadata` instance instead?
I can open a PR doing this against your branch or make another PR once this one is merged